tests/integration/test_target_snowflake.py

```diff
@@ -413,10 +405,58 @@ def test_column_name_change(self):
                 {'C_INT': 2, 'C_PK': 2, 'C_TIME': datetime.time(7, 15), 'C_VARCHAR': '2', 'C_TIME_RENAMED': None},
-                {'C_INT': 3, 'C_PK': 3, 'C_TIME': datetime.time(23, 0, 3), 'C_VARCHAR': '3', 'C_TIME_RENAMED': datetime.time(8, 15)},
+                {'C_INT': 3, 'C_PK': 3, 'C_TIME': datetime.time(23, 0, 3), 'C_VARCHAR': '3',
+                 'C_TIME_RENAMED': datetime.time(8, 15)},
                 {'C_INT': 4, 'C_PK': 4, 'C_TIME': None, 'C_VARCHAR': '4', 'C_TIME_RENAMED': datetime.time(23, 0, 3)}
             ])
 
+    def test_logical_streams_from_pg(self):
+        """Tests logical streams from pg with inserts, updates and deletes"""
+        tap_lines = test_utils.get_test_tap_lines('messages-logical-streams.json')
+
+        # Turning on hard delete mode
+        self.config['hard_delete'] = True
+        self.persist_lines_with_cache(tap_lines)
+
+        # Get loaded rows from tables
+        snowflake = DbSync(self.config)
+        target_schema = self.config.get('default_target_schema', '')
+        table_one = snowflake.query("SELECT * FROM {}.logical1_table1 ORDER BY cid".format(target_schema))
+        table_two = snowflake.query("SELECT * FROM {}.logical1_table2 ORDER BY cid".format(target_schema))
+        table_three = snowflake.query("SELECT * FROM {}.logical2_table1 ORDER BY cid".format(target_schema))
+
+        # ----------------------------------------------------------------------
+        # Check rows in table_one
+        # ----------------------------------------------------------------------
+        expected_table_one = [
+            {'cid': 1, 'cvarchar': "inserted row", 'cvarchar2': None},
+            {'cid': 2, 'cvarchar': 'inserted row', "cvarchar2": "inserted row"},
+            {'cid': 3, 'cvarchar': "inserted row", 'cvarchar2': "inserted row"},
+            {'cid': 4, 'cvarchar': "inserted row", 'cvarchar2': "inserted row"}
+        ]
+
+        # ----------------------------------------------------------------------
+        # Check rows in table_tow
+        # ----------------------------------------------------------------------
+        expected_table_two = [
+            {'cid': 1, 'cvarchar': "updated row"},
+            {'cid': 2, 'cvarchar': 'updated row'},
+            {'cid': 3, 'cvarchar': "updated row"},
+            {'cid': 5, 'cvarchar': "updated row"},
+            {'cid': 7, 'cvarchar': "updated row"},
+            {'cid': 8, 'cvarchar': 'updated row'},
+            {'cid': 9, 'cvarchar': "updated row"},
+            {'cid': 10, 'cvarchar': 'updated row'}
+        ]
+
+        # ----------------------------------------------------------------------
+        # Check rows in table_three
+        # ----------------------------------------------------------------------
+        expected_table_three = [
+            {'cid': 1, 'cvarchar': "updated row"},
+            {'cid': 2, 'cvarchar': 'updated row'},
+            {'cid': 3, 'cvarchar': "updated row"},
+        ]
```

Comment:
assertion seems to be missing here

Reply:
Yeah my bad, got distracted by Tech day and I forgot to add it.
I'm working on adding it.